R/SOS_200_methods-gda.R

```diff
@@ -65,15 +65,21 @@
 
   if (verbose) cat("[.getDataAvailability_1.0.0] REQUEST:\n", toString(.gda), "\n")
 
-  .responseString = sosRequest(sos = sos, request = .gda,
+  .responseString <- sosRequest(sos = sos, request = .gda,
                                verbose = verbose, inspect = inspect)
 
-  if (verbose) cat("[.getDataAvailability_1.0.0] Received response (size:", object.size(.responseString),
-                   "bytes), parsing ...\n")
-
-  if (verbose) cat("[.getDataAvailability_1.0.0] Response string: \n'", .responseString, "'\n")
+  if (verbose) {
+    cat("[.getDataAvailability_1.0.0] Received response (size:", object.size(.responseString),
+        "bytes), parsing ...\n")
+    cat("[.getDataAvailability_1.0.0] Response string: \n'", .responseString, "'\n")
+  }
 
-  if (isXMLString(.responseString)) {
+  if (.isEmptyGetDataAvailabilityResponse(.responseString)) {
+    if(verbose)
+      cat("[.getDataAvailability_1.0.0] Received empty data availability response.", "\n")
+    return(list())
+  }
+  else if (isXMLString(.responseString)) {
```

Comment:
Function `isXmlString()` does not exist anymore - can you rebase this feature branch to `upstream/feature/0.4` ?